packages/types/src/options.ts

```diff
@@ -144,4 +145,13 @@ export interface Options {
    * @returns The breadcrumb that will be added | null.
    */
   beforeBreadcrumb?(breadcrumb: Breadcrumb, hint?: BreadcrumbHint): Breadcrumb | null;
+
+  /**
+   * Function to compute tracing sample rate dynamically and filter unwanted traces.
+   *
+   * Can be defined in place of `tracesSampleRate`. If neither is defined, tracing is disabled.
+   *
+   * @returns A sample rate or `null` to drop the trace.
+   */
+  tracesSampler?(traceContext: SampleContext): number | null;
```

Comment:
Tbh I'm not sure why we accept `null` here. It makes things unnecessarily more complicated.
`0` is exactly the same as `null` from technical standpoint here.

Reply:
It's true - it's not totally necessary. I was going for the parallel with `beforeSend`, but if you think that's not worth it, I can take it out.

Reply:
Update: Conversation elsewhere has convinced me you're right. Will take it out.